File: jflove-app/lib/services/config_service.dart
import '../utils/http_service.dart';

/// 配置查询服务
/// 对标 jflove-desktop/src/services/config_service.py
class ConfigService {
  final HttpService _http;

  ConfigService(this._http);

  /// 查询服务端配置
  /// 对应服务端：GET /api/v1/config（需 admin 权限）
  Future<Map<String, dynamic>> getConfig() async {
    return _http.encryptedGet('/api/v1/config', {});
  }

  /// 更新服务端配置项
  /// 对应服务端：PUT /api/v1/config（需 admin 权限，写后立即生效、无需重启）
  Future<Map<String, dynamic>> updateConfig(String key, String value) async {
    return _http.encryptedPut('/api/v1/config', {'key': key, 'value': value});
  }
}

File: jflove-web/src/services/config-service.ts
/**
 * 服务端配置服务
 *
 * 对标桌面端 src/services/config_service.py。
 * 后端 `GET/PUT /api/v1/config` 均为管理员接口（加密 body + JWT）。
 */

import { httpClient } from '../utils/http-client';
import type { ServerConfig } from '../types/models';

/** 媒体修复相关配置键（与后端 settings.py 保持一致）
 * v1.4.2 hotfix：media_repair_enabled（实时修复总开关）与
 * media_repair_allow_transcode（重编码降级）均已废弃，前端不再读写 */
export const MEDIA_REPAIR_KEYS = {
  maxConcurrent: 'media_repair_max_concurrent',
} as const;

export const configService = {
  /** 获取服务端全部配置（key-value 映射，管理员） */
  async getConfig(): Promise<ServerConfig> {
    return httpClient.get<ServerConfig>('/api/v1/config');
  },

  /** 更新单个配置项（管理员，写后立即生效、无需重启） */
  async updateConfig(key: string, value: string): Promise<{ message: string }> {
    return httpClient.put<{ message: string }>('/api/v1/config', { key, value });
  },
};
